src/storage-engine/loader-helpers.ts

```diff
@@ -46,8 +42,10 @@ export async function parseCarFile(reader: CarReader): Promise<CarHeader> {
   const header = await reader.get(roots[0]);
   if (!header) throw new Error("missing header block");
   const { value } = await decode({ bytes: header.bytes, hasher, codec });
-  // @ts-ignore
-  if (value && value.fp === undefined) throw new Error("missing fp");
-  const { fp } = value as { fp: CarHeader };
-  return fp;
+  const fpvalue = value as { readonly fp: CarHeader };
+  // @jchris where is the fp attribute coming from?
```

Comment:
the `CarHeader` type name is misleading -- it's not a CAR file thing, it's just the data structure we put in the "root" block of the CAR file. Maybe it should be called `FireproofHeader`.

comes from 

```
export async function encodeCarHeader(fp: CarHeader) {
  return (await encode({
    value: { fp },
    hasher,
    codec
  })) as AnyBlock
}
```

the purpose is to make it clear this is a FireproofHeader, and not some other applications, when we load it from the storage engine.